core/src/test/scala/com/yahoo/maha/core/query/presto/PrestoQueryGeneratorTest.scala

```diff
@@ -347,4 +347,32 @@ ORDER BY mang_impressions ASC
     result should equal (expected) (after being whiteSpaceNormalised)
   }
 
+  test("generating presto query with outer group should fail") {
+    val jsonString =
+      s"""{
+          "cube": "s_stats",
+          "selectFields": [
+              {"field": "Campaign Name"},
+              {"field": "Average CPC"},
+              {"field": "Average Position"},
+              {"field": "Impressions"}
+          ],
+          "filterExpressions": [
+              {"field": "Advertiser ID", "operator": "=", "value": "12345"},
+              {"field": "Day", "operator": "between", "from": "$fromDate", "to": "$toDate"}
+          ]
+      }"""
+
+    val request: ReportingRequest = getReportingRequestAsync(jsonString)
+    val registry = getDefaultRegistry()
+    val requestModel = RequestModel.from(request, registry)
+
+    assert(requestModel.isSuccess, requestModel.errorMessage("Building request model failed"))
+
+
+    val queryPipelineTry = generatePipelineForQgenVersion(registry, requestModel.toOption.get, Version.v2)
```

Comment:
you can remove line and generate query on against v0. Theres no V2 version of Presto generator